parsec/core/types/manifest.py

```diff
@@ -266,14 +266,19 @@ def evolve_and_mark_updated(self: LocalManifestTypeVar, **data) -> LocalManifest
 
     @classmethod
     def from_remote(
-        cls: Type[LocalManifestTypeVar], remote: BaseRemoteManifest
+        cls: Type[LocalManifestTypeVar],
+        remote: BaseRemoteManifest,
+        pattern_filter: Optional[Pattern] = None,
+        local_manifest: Optional["BaseLocalManifest"] = None,
```

Comment:
It's kind of weird to create a `BaseLocalManifest` while providing another `BaseLocalManifest`... (this is especially a trouble given we use only part of the provided `local_manifest`, without explicitly saying so in the method signature)

It would be better to use `evolve` (or to create a specialized `evolve_...` method), but I guess the trouble is we are in a kind of middle ground here given we keep keep only part of the local_manifest (so the `evolve_...` method would reset the `need_sync` and `children` attributes, so not really an evolve...)

In the end I think the trouble is we are trying to put part of the local/remote merge logic in the manifest class... wouldn't it be better to only use `from_remote `/ `evolve` and keep the more complex logic in the `sync_transaction.merge_manifests` function ?